juju/juju
# Remove stand-alone unit agent concerns

2.9 introduced the embedded unit agent, run by the machiner.

Since we no longer run a stand-alone agent, this logic can be deleted along with associated dependency engine manifold concerns.

## QA steps

- Bootstrap 2.8 and deploy a workload (I used Ubuntu).
- Upgrade the controller, then model, to this patch with `--build-agent`.
- Wait for completion.
- `juju run --machine 0 "sudo grep upgraded /var/lib/juju/agents/unit-ubuntu-0/agent.conf"` should yield:
```
upgradedToVersion: 2.9.4.1
```
- `juju run --unit ubuntu/0 "network-get juju-info --format yaml"` should work:
```
bind-addresses:
- mac-address: 00:16:3e:ed:94:ec
  interface-name: eth0
  addresses:
  - hostname: ""
    address: 10.161.87.100
    cidr: 10.161.87.0/24
  macaddress: 00:16:3e:ed:94:ec
  interfacename: eth0
egress-subnets:
- 10.161.87.100/32
ingress-addresses:
- 10.161.87.100
```

## Documentation changes

None.

## Bug reference

N/A
